spec/spec.md

```diff
@@ -302,46 +306,53 @@ authorize
 
 ###### Errors
 {: .no_toc }
 
 - `-32602` (Invalid params) if the params object does not match the format defined above
 - `ERROR_AUTHORIZATION_FAILED` if the wallet endpoint did not authorize access to the requested privileged methods
+- `ERROR_CLUSTER_NOT_SUPPORTED` if the wallet endpoint does not support the requested Solana cluster
 
 ##### Description
-This method allows the dapp endpoint to request authorization from the wallet endpoint for access to [privileged methods](#privileged-methods). On success, it returns an `auth_token` providing access to privileged methods. It may also return a URI suitable for future use as an [endpoint-specific URI](#endpoint-specific-uris).
+
+This method allows the dapp endpoint to request authorization from the wallet endpoint for access to [privileged methods](#privileged-methods). On success, it returns an `auth_token` providing access to privileged methods, along with addresses for all authorized accounts. It may also return a URI suitable for future use as an [endpoint-specific URI](#endpoint-specific-uris).
 
 The returned `auth_token` is an opaque string with meaning only to the wallet endpoint which created it. It is recommended that the wallet endpoint include a mechanism to authenticate the contents of auth tokens it issues (for e.g., with an HMAC, or by encryption with a secret symmetric key).
 
 The lifetime of the returned `auth_token` is not defined. If a privileged method returns `ERROR_REAUTHORIZE`, the dapp endpoint should call the [`reauthorize`](#reauthorize) method to renew the token.
 
-Dapp endpoints should make every effort possible to [verify the authenticity](#dapp-identity-verification) of the presented identity. While the `dapp_uri` parameter is optional, it is strongly recommended - without it, the wallet endpoint may not be able to verify the authenticity of the dapp.
+Dapp endpoints should make every effort possible to [verify the authenticity](#dapp-identity-verification) of the presented identity. While the `uri` parameter is optional, it is strongly recommended - without it, the wallet endpoint may not be able to verify the authenticity of the dapp.
+
+The `cluster` parameter allows the dapp endpoint to select a specific Solana cluster with which to interact. This is relevant for both [`sign_transactions`](#sign_transactions), where a wallet may refuse to sign transactions without a currently valid blockhash, and for [`sign_and_send_transactions`](#sign_and_send_transactions), where the wallet endpoint must know which cluster to submit the transactions to. This parameter would normally be used to select a cluster other than `mainnet-beta` for dapp development and testing purposes. Under normal circumstances, this field should be omitted, in which case the wallet endpoint will interact with the `mainnet-beta` cluster.
```

Comment:
I wonder; do we actually expect wallets to check the validity of a blockhash before signing, or is that not typical. I guess the words ‘may refuse’ covers this. Maybe make that even looser via ‘may choose to refuse?’

Or maybe take it out altogether; if I pass a wallet a transaction that includes a _durable nonce_ value, there's nothing that they can do to verify it's still valid unless I _also_ pass them the address of the nonce account.

Just trying to prevent the spec from writing cheques it can't cash here.

Reply:
Wallets should probably just shut up and sign transactions. Maybe we should take `cluster` out here.

Reply:
cc/ @jordansexton.

Reply:
> Wallets should probably just shut up and sign transactions

The wallet needs to know this, so it knows where to send a transaction. I think the only debate is whether this should be a property of `sign_and_send_transaction`, or of `authorize`. Putting it here allows the wallet to make better decisions (but does not force it to do so).

Reply:
Oh no, I wasn't talking about `sign_and_send_transaction`. I was talking about `sign_transaction`. You can definitely sign without knowing the cluster.

Reply:
It's this bit that I'm stuck on.

> This is relevant for both [`sign_transactions`](#sign_transactions), where a wallet may refuse to sign transactions without a currently valid blockhash

Reply:
That was a suggestion I got from @jordansexton. Let's wait for him to weigh in.